nestkernel/simulation_manager.cpp

```diff
@@ -410,11 +466,19 @@ nest::SimulationManager::simulate( Time const& t )
     LOG( M_ERROR, "SimulationManager::simulate", msg );
     throw KernelException();
   }
+}
+
+void
+nest::SimulationManager::run( Time const& t )
+{
+  check_run( t );
 
   to_do_ += t.get_steps();
   to_do_total_ = to_do_;
+  assert( to_do_ != 0 );
 
-  const size_t num_active_nodes = prepare_simulation_();
+  // Check whether waveform relaxation is used on any MPI process
+  kernel().node_manager.check_wfr_use();
```

Comment:
That's where `prepare_simulation_` was, the only part surviving is the check_wfr_use call. Maybe it could happen only once --- but I'm not sure, not tested.